parsec/mca/device/transfer_gpu.c

```diff
@@ -239,10 +239,19 @@ parsec_gpu_create_w2r_task(parsec_device_gpu_module_t *gpu_device,
             break;
         }
         gpu_copy = (parsec_gpu_data_copy_t*)item;
+        cpu_copy = gpu_copy->original->device_copies[0];
         parsec_atomic_lock( &gpu_copy->original->lock );
         /* get the next item before altering the next pointer */
         item = (parsec_list_item_t*)item->list_next;  /* conversion needed for volatile */
-        if( 0 == gpu_copy->readers ) {
+        if (cpu_copy->flags & PARSEC_DATA_FLAG_DISCARDED) {
+            parsec_list_item_ring_chop((parsec_list_item_t*)gpu_copy);
```

Comment:
And here.

Reply:
Same as above. The flag isn't changed and the device copy maintains a reference on the data_t so if we miss the update of the flags here we will evict into the host copy and then release everything.